ast/check.go

```diff
@@ -705,13 +704,14 @@ func (rc *refChecker) checkRef(curr *TypeEnv, node *typeTreeNode, ref Ref, idx i
 			return nil
 		}
 
-		if !unifies(types.S, exist) {
+		if !unifies(types.S, exist) { // NOTE(sr): types.S ok here?
```

Comment:
Until this function reaches a leaf in the type tree, we're assuming that all of the keys must be strings because these correspond to elements in the package path.